packages/@react-stately/dnd/src/useDroppableCollectionState.ts

```diff
@@ -25,42 +25,82 @@ export interface DroppableCollectionState {
   target: DropTarget,
   setTarget(target: DropTarget): void,
   isDropTarget(target: DropTarget): boolean,
-  getDropOperation(target: DropTarget, types: DragTypes, allowedOperations: DropOperation[]): DropOperation
+  getDropOperation(e: DropOperationEvent): DropOperation
 }
 
 export function useDroppableCollectionState(props: DroppableCollectionStateOptions): DroppableCollectionState  {
+  let {
+    acceptedDragTypes,
+    onInsert,
+    onRootDrop,
+    onItemDrop,
+    onReorder,
+    shouldAcceptItemDrop,
+    collection,
+    selectionManager,
+    onDropExit,
+    onDropEnter,
+    getDropOperation,
+    onDrop
+  } = props;
   let [target, setTarget] = useState<DropTarget>(null);
 
   let getOppositeTarget = (target: ItemDropTarget): ItemDropTarget => {
     if (target.dropPosition === 'before') {
-      let key = props.collection.getKeyBefore(target.key);
+      let key = collection.getKeyBefore(target.key);
       return key != null ? {type: 'item', key, dropPosition: 'after'} : null;
     } else if (target.dropPosition === 'after') {
-      let key = props.collection.getKeyAfter(target.key);
+      let key = collection.getKeyAfter(target.key);
       return key != null ? {type: 'item', key, dropPosition: 'before'} : null;
     }
   };
 
+  let defaultGetDropOperation = useCallback((e: DropOperationEvent) => {
+    let {
+      target,
+      types,
+      allowedOperations,
+      isInternalDrop,
+      draggingKeys
+    } = e;
+
+    if (acceptedDragTypes === 'all' || acceptedDragTypes.some(type => types.has(type))) {
+      let isValidInsert = onInsert && target.type === 'item' && !isInternalDrop && (target.dropPosition === 'before' || target.dropPosition === 'after');
+      // TODO: added allowedOperations[0] !== 'copy' to block internal copy insert operations as per feedback. Perhaps rename onReorder to onInternalInsert and remove this restriction? Maybe this is just ok?
```

Comment:
I think we just remove this restriction and let onReorder handle both move and copy.

Reply:
Sounds good, I'm curious what you think about the naming of `onReorder` in that case? Maybe a bit confusing, not sure if an internal copy operation is something a user would think of when writing `onReorder`? `onInternalInsert` feels a bit much and we probably would then need to name `onInsert` to `onExternalInsert`...

Reply:
For posterity: we decided to stick with onReorder since it felt like it reflected the most common use case